File: Iv Layhort/Java_Layhort/src/Task_6/Exercise_1_DrawingBook.java
import java.util.*;
public class Exercise_1_DrawingBook {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int numberOfPages,destinationPage;
        System.out.println("Enter the number of pages and destination page respectively :");
        numberOfPages = input.nextInt();
        destinationPage = input.nextInt();

        System.out.println((pageCount(numberOfPages,destinationPage))==0 ? "No need to turn" : "The minimum turn is"+pageCount(numberOfPages,destinationPage));
    }

    private static int pageCount(int numberOfPages,int destinationPage){
        int totalPageTurnCountFromFront,targetPageCountFromFront,targetPageCountFromBack;
        totalPageTurnCountFromFront = numberOfPages/2;
        targetPageCountFromFront = destinationPage/2;
        targetPageCountFromBack = totalPageTurnCountFromFront - targetPageCountFromFront;

        return Math.min(targetPageCountFromFront,targetPageCountFromBack);
    }
}

File: Iv Layhort/Scala-Layhort/src/Task_6/Exercise_1_DrawingBook.scala
package Task_6
import scala.io.StdIn._
object Exercise_1_DrawingBook {
  def main(args: Array[String]): Unit = {
    println("Enter the number of pages and destination page respectively :")
    val Array(numberOfPages,destinationPage) = readLine().split(" ").map(_.toInt)
    println(if(pageTurn(numberOfPages,destinationPage)==0) "No need to turn" else s"You have to turn ${pageTurn(numberOfPages,destinationPage)} time/times)
  }
  def pageTurn(numberOfPages:Int,destinationPage:Int):Int={
    val totalPageTurnCountFromFront = numberOfPages/2
    val targetPageTurnCountFromFront = destinationPage/2
    val targetPageTurnCountFromBack = totalPageTurnCountFromFront-targetPageTurnCountFromFront

    Math.min(targetPageTurnCountFromFront,targetPageTurnCountFromBack)
  }
}
